### ISSUE TITLE
Editor crashes after deleting root level prefab entity alongside other entities

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#5782]
**Describe the bug**
When the level prefab is selected with other entities, it can be deleted alongside them, which crashes the Editor.

**Steps to reproduce**
1. Create a new level.
2. Select all entities.
3. Press the **Delete** key or select Delete from the context menu.

**Expected behavior**
The level prefab cannot be deleted.

**Actual behavior**
Editor crashes after deleting the level prefab.

**Visual Studio Call stack Trace:**
```
 	000001acfd0d1f90()	Unknown
>	EditorLib.dll!AZ::JsonSerializer::ResolvePointer(AZ::JsonSerializationResult::ResultCode & status, AZ::JsonSerializer::StoreTypeId & storeTypeId, const void * & object, const void * & defaultObject, AZStd::any & defaultObjectStorage, const AZ::SerializeContext::ClassData * & elementClassData, const AZ::IRttiHelper & rtti, AZ::JsonSerializerContext & context) Line 494	C++
 	EditorLib.dll!AZ::JsonSerializer::StoreWithClassDataFromPointer(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::SerializeContext::ClassData & classData, AZ::JsonSerializer::UseTypeSerializer custom, AZ::JsonSerializerContext & context) Line 172	C++
 	EditorLib.dll!AZ::JsonSerializer::StoreFromPointer(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializer::UseTypeSerializer custom, AZ::JsonSerializerContext & context) Line 92	C++
 	EditorLib.dll!AZ::BaseJsonSerializer::ContinueStoring(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializerContext & context, AZ::BaseJsonSerializer::ContinuationFlags flags) Line 262	C++
 	EditorLib.dll!AZ::JsonSmartPointerSerializer::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & outputValue, const void * inputValue, const void * defaultValue, const AZ::Uuid & valueTypeId, AZ::JsonSerializerContext & context) Line 156	C++
 	EditorLib.dll!AZ::JsonSerializer::StoreWithClassData(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & node, const void * object, const void * defaultObject, const AZ::SerializeContext::ClassData & classData, AZ::JsonSerializer::StoreTypeId storeTypeId, AZ::JsonSerializer::UseTypeSerializer custom, AZ::JsonSerializerContext & context) Line 136	C++
 	EditorLib.dll!AZ::JsonSerializer::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializer::UseTypeSerializer custom, AZ::JsonSerializerContext & context) Line 70	C++
 	EditorLib.dll!AZ::BaseJsonSerializer::ContinueStoring(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializerContext & context, AZ::BaseJsonSerializer::ContinuationFlags flags) Line 262	C++
 	EditorLib.dll!AZ::BaseJsonSerializer::ContinueStoringToJsonObjectField(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, rapidjson_ly::GenericStringRef<char> newMemberName, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializerContext & context, AZ::BaseJsonSerializer::ContinuationFlags flags) Line 322	C++
 	EditorLib.dll!AzToolsFramework::Prefab::JsonInstanceSerializer::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & outputValue, const void * inputValue, const void * defaultValue, const AZ::Uuid & valueTypeId, AZ::JsonSerializerContext & context) Line 59	C++
 	EditorLib.dll!AZ::JsonSerializer::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, const void * object, const void * defaultObject, const AZ::Uuid & typeId, AZ::JsonSerializer::UseTypeSerializer custom, AZ::JsonSerializerContext & context) Line 42	C++
 	EditorLib.dll!AZ::JsonSerialization::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> & output, rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator> & allocator, const void * object, const void * defaultObject, const AZ::Uuid & objectType, AZ::JsonSerializerSettings & settings) Line 322	C++
 	[Inline Frame] EditorLib.dll!AZ::JsonSerialization::Store(rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>> &) Line 345	C++
 	EditorLib.dll!AzToolsFramework::Prefab::PrefabDomUtils::StoreInstanceInPrefabDom(const AzToolsFramework::Prefab::Instance & instance, rapidjson_ly::GenericDocument<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>,rapidjson_ly::CrtAllocator> & prefabDom, AzToolsFramework::Prefab::PrefabDomUtils::StoreFlags flags) Line 109	C++
 	EditorLib.dll!AzToolsFramework::Prefab::PrefabPublicHandler::DeleteFromInstance(const AZStd::vector<AZ::EntityId,AZStd::allocator> & entityIds, bool deleteDescendants) Line 1244	C++
 	EditorLib.dll!AzToolsFramework::Prefab::PrefabPublicHandler::DeleteEntitiesAndAllDescendantsInInstance(const AZStd::vector<AZ::EntityId,AZStd::allocator> & entityIds) Line 984	C++
 	[Inline Frame] ComponentEntityEditorPlugin.dll!AzToolsFramework::Prefab::PrefabIntegrationManager::ContextMenu_DeleteSelected() Line 637	C++
 	[Inline Frame] ComponentEntityEditorPlugin.dll!AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu::__l2::<lambda_7>::operator()() Line 386	C++
 	[Inline Frame] ComponentEntityEditorPlugin.dll!QtPrivate::FunctorCall<QtPrivate::IndexesList<>,QtPrivate::List<>,void,`AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu'::`2'::<lambda_7>>::call(AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu::__l2::<lambda_7> &) Line 146	C++
 	[Inline Frame] ComponentEntityEditorPlugin.dll!QtPrivate::Functor<`AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu'::`2'::<lambda_7>,0>::call(AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu::__l2::<lambda_7> &) Line 256	C++
 	ComponentEntityEditorPlugin.dll!QtPrivate::QFunctorSlotObject<`AzToolsFramework::Prefab::PrefabIntegrationManager::PopulateEditorGlobalContextMenu'::`2'::<lambda_7>,0,QtPrivate::List<>,void>::impl(int which, QtPrivate::QSlotObjectBase * this_, QObject * r, void * * a, bool * ret) Line 443	C++
 	[External Code]	
 	ComponentEntityEditorPlugin.dll!AzToolsFramework::EntityOutlinerWidget::OnOpenTreeContextMenu(const QPoint & pos) Line 639	C++
 	[External Code]	
 	EditorLib.dll!CryEditMain(int argc, char * * argv) Line 4190	C++
 	[Inline Frame] Editor.exe!AZStd::Internal::INVOKE(int(*)(int, char * *) &) Line 208	C++
 	[Inline Frame] Editor.exe!AZStd::invoke(int(*)(int, char * *) &) Line 38	C++
 	Editor.exe!main(int argc, char * * argv) Line 32	C++
 	Editor.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Line 97	C++
 	[External Code]
```	

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/142604349-4c454744-7f11-4101-b5d5-c06d13f4576a.mp4

![exception](https://user-images.githubusercontent.com/86952009/142604356-113f9229-833f-4811-b670-f54f139a4d57.JPG)

**Found in Branch**
Stabilization_2110 (https://github.com/o3de/o3de/commit/d30fdb759d07e54d36309680ffa4247f1481ceb2)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

**Additional context**
[dump.zip](https://github.com/o3de/o3de/files/7569450/dump.zip)
[Editor.log](https://github.com/o3de/o3de/files/7569452/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7569453/error.log)

**Comment by AMZN-daimini** [comment]
This is a regression introduced by https://github.com/o3de/o3de/pull/5762
I have fixed the oversight and will have a PR for this soon. Thank you for finding this :D

**Comment by monroegm** [comment]
Fixed by #5794

**Comment by jckand-amzn** [comment]
Verified fix on stabilization/2110@3cf3e45b38d6626.

**Comment by LB-BartoszDudzinski** [comment]
Verified as Fixed in Stabilization
